### PR TITLE
fix: [AAP-23488] disallow deletion of "worker offline" activation

### PR DESCRIPTION
**Pull request by jshimkus-rh**

### CODE REVIEW

File: src/aap_eda/api/views/activation.py [hunk 1]
```diff
@@ -110,6 +99,39 @@ def create(self, request):
...
+                    "cannot be deleted."
+                ),
+            ),
+        },
+    )
+    def destroy(self, request, *args, **kwargs):
+        activation = self.get_object()
+
+        if activation.status == ActivationStatus.WORKERS_OFFLINE:
+            return Response(status=status.HTTP_409_CONFLICT)
```
Comment (Alex-Izquierdo) [inline src/aap_eda/api/views/activation.py:123]: I think here we should have a explicit error message.

Comment (jshimkus-rh) [reply src/aap_eda/api/views/activation.py:123]: Explicit how?  Include the activation id? name?
A log entry?

Comment (Alex-Izquierdo) [reply src/aap_eda/api/views/activation.py:123]: I mean we must say the reason. Example: https://github.com/ansible/eda-server/blob/main/src/aap_eda/api/views/eda_credential.py#L249

Comment (jshimkus-rh) [reply src/aap_eda/api/views/activation.py]: Yep.  Figured that out. 😄

**Review by ttuffin — APPROVED** [review]

**Review by Alex-Izquierdo — APPROVED** [review]

**Pull request merged**